include/ghex/structured/cubed_sphere/field_descriptor.hpp

```diff
@@ -0,0 +1,504 @@
+                            // add offset
+                            const coordinate_type x = local_coordinate + unpack_is.m_data_is.m_first;
+                            // assign
+                            unpack_is.data(x) = unpack_is.buffer(x);
+                        }
+                    }
+                }
+
+                template<typename LMap>
+                struct serialization<::gridtools::ghex::gpu,LMap> {
+                    // kernels use a 1-D compute grid with block-dim threads per block
+                    static constexpr std::size_t block_dim = 128;
+                    // number of values to pack/unpack per thread
+                    static constexpr std::size_t elements_per_thread = 1;
+
+                    template<typename PackIterationSpace>
+                    static void pack(PackIterationSpace&& pack_is, void* arg) {
+                        using coordinate_type = typename PackIterationSpace::coordinate_t;
+                        auto stream_ptr = reinterpret_cast<cudaStream_t*>(arg);
+                        const std::size_t num_threads = (pack_is.m_buffer_desc.m_size + elements_per_thread-1)
+                            /elements_per_thread;
+                        const std::size_t num_blocks = (num_threads+block_dim-1)/block_dim;
+                        pack_kernel<LMap><<<num_blocks,block_dim,0,*stream_ptr>>>(pack_is, elements_per_thread);
+                    }
+
+                    template<typename UnPackIterationSpace>
+                    static void unpack(UnPackIterationSpace&& unpack_is, void* arg) {
+                        using coordinate_type = typename UnPackIterationSpace::coordinate_t;
+                        auto stream_ptr = reinterpret_cast<cudaStream_t*>(arg);
+                        const std::size_t num_threads = (unpack_is.m_buffer_desc.m_size + elements_per_thread-1)
+                            /elements_per_thread;
+                        const std::size_t num_blocks = (num_threads+block_dim-1)/block_dim;
+                        unpack_kernel<LMap><<<num_blocks,block_dim,0,*stream_ptr>>>(unpack_is, elements_per_thread);
+                    }
+                };
+#endif
+
+                /** @brief Describes a field spanning a subdomain of a cubed-sphere tile
+                  * @tparam T value type
+                  * @tparam Arch Architecture type
+                  * @tparam ComponentOrder Data layout order parameter (3 == stride 1, 0 == largest stride)
+                  * @tparam XOrder Data layout order parameter (3 == stride 1, 0 == largest stride)
+                  * @tparam YOrder Data layout order parameter (3 == stride 1, 0 == largest stride)
+                  * @tparam ZOrder Data layout order parameter (3 == stride 1, 0 == largest stride)*/
+                template<typename T, typename Arch, int ComponentOrder=0, int XOrder=3, int YOrder=2, int ZOrder=1>
```

Comment:
why not using `layout_map`? 
This my default is for a 4D storage?

Reply:
changed this to variadic template parameter pack - 3D and 4D possible